backend/server.js

```diff
@@ -23,7 +64,41 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+//Registration endpoint where name, email and password can be assigned to a User in the database 
+app.post('/users', async (req, res) => {
+  try {
+    const { name, email, password } = req.body
+    //TO NOT STORE PLAIN TEXT PASSWORDS
+    const user = new User({ name, email, password: bcrypt.hashSync(password) })
+    const saved = await
+      // const saved = await user.save() res.status(201).json(saved)
+      user.save()
+    res.status(201).json({ id: user._id, accessToken: user.accessToken, message: "✨Created user ✨" })
+  } catch (err) {
+    res.status(400).json({ message: 'Could not create User', errors: err.errors })
+  }
+})
+
+//Secret endpoint
+//Applies the middleware-function above that checks authentication
+app.get('/secrets', authenticateUser)
+app.get('/secrets', (req, res) => {
+  res.json({ message: '"And above all, watch with glittering eyes the whole world around you because the greatest secrets are always hidden in the most unlikely places. Those who don´t believe in magic will never find it."' })
+})
+
+//Logging in endpoint
+app.post('/sessions', async (req, res) => {
+  const user = await User.findOne({ email: req.body.email })
+  //cleartext password from request compared to hashed password in database
+  if (user && bcrypt.compareSync(req.body.password, user.password)) {
+    res.json({ userId: user._id, accessToken: user.accessToken })
+  }
```

Comment:
When registering, we return `{id: user._id...`  Is there any reason that the `userId` in the response should change name here?